src/components/button/_macro.spec.js

```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
+                    renderComponent('button', {
+                        url: 'http://example.com',
+                        variants: 'download',
+                    }),
+                );
+                expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
+            });
+            it('THEN: renders download icon', () => {
+                const faker = templateFaker();
+                const iconsSpy = faker.spy('icon');
 
-        it('is an `button` element', () => {
-            const $ = cheerio.load(renderComponent('button'));
+                faker.renderComponent('button', {
+                    url: 'http://example.com',
+                    variants: 'download',
+                });
 
-            expect($('button').length).toBe(1);
+                expect(iconsSpy.occurrences[0].iconType).toBe('download');
+            });
+            it('THEN: renders the download attribute when variants contains download', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'download',
+                    }),
+                );
+
+                expect($('.ons-btn').attr('download')).toBeDefined();
+            });
         });
+        describe('WHEN: variants contains loader', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'loader',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
+            it('THEN: loader icon when variants contains loader', () => {
+                const faker = templateFaker();
+                const iconsSpy = faker.spy('icon');
 
-        it('has the provided `type` attribute', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    type: 'special-type',
-                }),
-            );
+                faker.renderComponent('button', {
+                    variants: 'loader',
+                });
 
-            expect($('button').attr('type')).toBe('special-type');
+                expect(iconsSpy.occurrences[0].iconType).toBe('loader');
+            });
         });
-
-        it('has the provided `type` attribute even if print variant is provided', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    type: 'special-type',
-                    variants: 'print',
-                }),
-            );
-
-            expect($('button').attr('type')).toBe('special-type');
+        describe('WHEN: variants contains timer', () => {
+            it('THEN: renders provided variant style classes', () => {
```

Comment:
```suggestion
            it('THEN: the button has the timer classes', () => {
```
Theres no classes provided, these are automatically set